plugins/modules/packaging/language/npm.py

```diff
@@ -216,15 +216,16 @@ def list(self):
             self.module.fail_json(msg="Failed to parse NPM output with error %s" % to_native(e))
         if 'dependencies' in data:
             for dep, props in data['dependencies'].items():
-                dep_version = dep + '@' + str(props['version'])
 
                 if 'missing' in props and props['missing']:
                     missing.append(dep)
                 elif 'invalid' in props and props['invalid']:
                     missing.append(dep)
                 else:
                     installed.append(dep)
-                    installed.append(dep_version)
+                    if props['version']:
```

Comment:
Is this supposed to be `if 'version' in props:` or will `version` always be present and it could be an empty string?

Reply:
Updated to check for `version` existence as well